packages/next/src/components/Yoast.tsx

```diff
@@ -12,8 +12,11 @@ import {
 import { useSettings } from '@headstartwp/core/react';
 import Head from 'next/head';
 
-function convertUrl(url: string, hostUrl: string, sourceUrl: string) {
-	return `${hostUrl}${removeSourceUrl({ link: url, backendUrl: sourceUrl })}`;
+export function convertUrl(url: string, hostUrl: string, sourceUrl: string) {
+	return `${hostUrl}${removeSourceUrl({
+		link: url.replace(/\/?$/, '/'),
+		backendUrl: sourceUrl,
+	})}`;
```

Comment:
@tobeycodes this won't resolve the issue we're seeing on Publix. This will ensure that the canonical has a trailing slash, but the issue we're seeing is that `hostUrl` is always appended. It should instead be conditional, based upon whether the input starts with the `sourceUrl`.

For example, assuming we have a `hostUrl` of `headless.test` and a `sourceUrl` of `backend.test` this is what we’d get for the canonicals.

| **Input**                          | **Current Output**                                      | **Expected Output**                |
| :--------------------------------- | :------------------------------------------------------ | :--------------------------------- |
| https://backend.test/some-page     | https://headless.test/some-page                         | https://headless.test/some-page    |
| https://headless.test/some-page    | https://headless.testhttps://headless.test/some-page    | https://headless.test/some-page    |
| https://external.test/another-page | https://headless.testhttps://external.test/another-page | https://external.test/another-page |

For context, this is an issue because Yoast allows editors to define their own canonical values, so we can't always assume that they'll point to a page on the backend.

Reply:
I have updated this to only work when the url starts with sourceUrl